New functionality and some bugfixes in analysis of nested archives

twogee opened the pull request:
I hope this is acceptable for a review; please let me know if you would like the changes to be split into patches for different themes (HTML generation, bundling of class files/component analysis, Java 5/style).

Commit:
Port to Java 5 (inspired by Ant 1.9)

Commit:
Refactored ArchiveType

Commit:
Code review

reviewed coding conventions and javadoc;
tattletale properties can be sed in Ant project instead of property
file;
fixed a related NPE in Configuration;
new take on TTALE-199; support for adapters (RAR) and partial extraction
of archives

Commit:
Analyze class file bundles as components etc

Added readme.txt
Changed some report names
Reviewed exceptions

Commit:
Bug fixes (analysis of nested archives, empty the ProcessBuilder buffers)

Commit:
Code review, always remove files in java.io.tmpdir

Commit:
Refactoring and/or bug fixes of Scanners and Reports

I had introduced a massive amount of duplications in Scanners;
Reports could not get paths in links straight, especially for nested
archives
(besides redundant code)

Commit:
Reread the comment to TTALE-213

It is better to clear floats rather than envelop the entire report body
in a paragraph
(I was confused by <p/> :-)

Commit:
Corrected markup

Commit:
Removed more copy-and-paste HTML generating code

Commit:
A teeny addition to readme.txt

jesperpedersen commented:
The best thing is to create a forum post which explains the changes before any development is done. So do that now, and we can leave the PR open until we find the changes needed. Thanks

Commit:
Configurable Ant property prefix

Commit:
Custom reports need only implement Report

Commit:
Revised documentation

Added new task attributes/properties
Revised report names
Added warning about old report names to Main for backwards compatibility

Commit:
Further refactoring of HTML generating code

Commit:
Severity/Status as classes are superfluous

Reports need not to extend Severity/Status because Main decides what
failOnInfo/Warning/Error is.
Otherwise, there must be a way to tell Main how to interpret the
semantics of extended Severity/Status,
which would unnecessarily complicate the API.

twogee committed:
checkstyle and toolchain adjustments

twogee committed:
Documentation update

twogee committed:
More checkstyle adjustments

twogee committed:
simplify

twogee committed:
correct filename extension

twogee committed:
update javassist

twogee committed:
readme formatted according to markdown spec

twogee committed:
A typo

twogee committed:
Added dependency graph reduction

twogee committed:
Make things immutable

twogee committed:
pretty print

twogee committed:
use the latest Ivy

twogee committed:
Use the latest Javassist/style checks

twogee committed:
Do not try to retrieve Ivy every time

twogee committed:
use entity name for copyright sign

twogee committed:
Target Java 6

twogee committed:
Use Java 6 language features

twogee committed:
Maven SDL

twogee committed:
Keeping checkstyle happy...

twogee committed:
Forgot to add the property for tred (see 2ce9fbb)

twogee committed:
Make doclint happy

twogee committed:
Add support for Java up to version 9;
keep dependencies compatible to Java 5

twogee committed:
Tidy up the code

twogee committed:
newInstance() is deprecated

twogee committed:
Simplify

twogee committed:
Laziness is a virtue ;-)

twogee committed:
Use the newest Java 6-compatible components

twogee committed:
Update to newest Java 6-compatible Checkstyle

twogee committed:
Fix indentation

twogee committed:
Indent XSLT

twogee committed:
Indent build files

twogee committed:
Use strict XHTML

twogee committed:
Indent documentation

twogee committed:
Use isEmpty()

twogee committed:
Simplify

twogee committed:
More simplifications

twogee committed:
DRY…

twogee committed:
Fix URL:s (prefer HTTPS), update to Ivy 2.5.0

twogee committed:
DRY again :-)

twogee committed:
Add Travis

twogee committed:
Update dependencies
